### PR TITLE
new codes

### PR DESCRIPTION
**Pull request by noahberthusen**
Here are my additions to the zoo for the CMSC 868Q assignment.

### CODE REVIEW

File: codes/classical/bits/binary_golay.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+code_id: binary_golay
+physical: bits
+logical: bits
+
+name: 'Binary Golay code'
+introduced: '\cite{manual:{Golay, M. J. E. "Notes on Digital Coding." Proc. IRE 37, 657, 1949.}}'
+
```
Comment (valbert4) [inline codes/classical/bits/binary_golay.yml:7]: [n,k,d] in \(\) latex quotes

File: codes/classical/bits/fountain.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  - 'Useful in situations like broadcasting and file distribution.'
+
+notes:
+  - 'Random linear fountain codes approach the Shannon limit as the file size \(K\) increases.'  
+
+relations:
+#  parents:
+#    - code_id: tornado
+  children:
+    - code_id: fountain
```
Comment (valbert4) [inline codes/classical/bits/fountain.yml:33]: There is not "children" entry; I've added this info to spreadsheet though. Need to add a Parent = largest code class in the same alphabet that contains the code. Could be bits_into_bits or something smaller.

File: codes/classical/bits/fountain.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
```
Comment (valbert4) [inline codes/classical/bits/fountain.yml:34]: Add cousin: random

File: codes/classical/bits/lt.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+name: 'Luby Transform (LT)'
+introduced: '\cite{doi:10.1109/SFCS.2002.1181950}'
+
+description: 'Luby Transform codes are erasure codes based on fountain codes. They improve on random linear fountain codes by having a much more efficient encoding and decoding algorithm.'
+
+protection: 'Designed to protect agains erasure errors. '
+
+features:
+  rate: 'Being fountain codes, LT codes do not have a fixed rate.'
```
Comment (valbert4) [inline codes/classical/bits/lt.yml:13]: No need for this statement.

File: codes/classical/bits/lt.yml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  decoders:
+    - 'Decoded using the sum-product algorithm \cite{doi:10.5555/971143}, similar to belief propagation \cite{doi:10.5555/2876686.2876719}.'
+
+realizations:
+  - 'Useful in broadcasting and file distribution.'
+
+relations:
+  parents:
+    - code_id: fountain
+  cousins:
```
Comment (valbert4) [inline codes/classical/bits/lt.yml:28]: cousin: random

File: codes/classical/bits/binary_golay.yml [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+realizations:
+  - 'Code was used in the Voyager 1 and 2 spacecraft \cite{doi:10.1029/JA092iA13p14873}.'
+
+notes:
+  - 'The automorphism group of the binary Golay code is the Mathieu group \(\mathcal{M}_{23}\) and the automorphism group of the extended binary Golay code is the Mathieu group \(\mathcal{M}_{24}\), two of the sporadic simple groups.'  
+
+relations:
+  parents:
+    - code_id: binary_linear
```
Comment (valbert4) [inline codes/classical/bits/binary_golay.yml:32]: @AlexanderBarg any cousins applicable here?

File: codes/classical/bits/binary_golay.yml [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+features:
+  rate: 'The perfect binary Golay code has a rate of 12/23 = 0.522. The extended binary Golay code has a rate of 12/24 = 0.5.'
+  
+  encoders:
+    - 'Can be constructed with the polynomial \(x^{11} + x^{10} + x^6 + x^5 + x^4 + x^2 + 1\) over GF(2)'
+    - 'One can use the Great Dodecahedron to generate codewords by placing message bits on the faces and calculating the parity bits that live on the 12 vertices of the icosahedron'
+
+  decoders:
+    - 'Since \(n\) is a constant, table-lookup decoding is a possibility.'
```
Comment (AlexanderBarg) [inline codes/classical/bits/binary_golay.yml:21]: Mention also trellis decoding.

File: codes/classical/bits/binary_golay.yml [hunk 8]
```diff
@@ -0,0 +1,33 @@
+code_id: binary_golay
```
Comment (valbert4) [inline codes/classical/bits/binary_golay.yml:1]: golay

**Pull request closed without merging**